src/Hazelcast.Net/Clustering/RetryStrategy.cs

```diff
@@ -71,19 +72,39 @@ public RetryStrategy(string action, int initialBackOffMilliseconds, int maxBackO
             _action = action.ToLowerInvariant();
 #pragma warning restore CA1308
 
-
+            if (initialBackOffMilliseconds < 0) throw new ConfigurationException("Initial back-off must be greater than or equal to zero.");
             _initialBackoffMilliseconds = initialBackOffMilliseconds;
+            if (maxBackOffMilliseconds < 0) throw new ConfigurationException("Maximum back-off must be greater than or equal to zero.");
             _maxBackoffMilliseconds = maxBackOffMilliseconds;
+            if (multiplier <= 0) throw new ConfigurationException("Multiplier must be greater than zero.");
             _multiplier = multiplier;
             _timeoutMilliseconds = timeoutMilliseconds;
+            if (jitter < 0 || jitter > 1) throw new ConfigurationException("Jitter must be between zero and one, inclusive.");
             _jitter = jitter;
-            _currentBackOffMilliseconds = initialBackOffMilliseconds;
 
             _logger = loggerFactory?.CreateLogger<RetryStrategy>() ?? throw new ArgumentNullException(nameof(loggerFactory));
 
             Restart();
         }
 
+        /// <summary>
+        /// (internal for tests only) Gets the delay.
+        /// </summary>
+        internal int GetDelay(int elapsed)
+        {
+            var delay = (int)(_currentBackOffMilliseconds * (1 - _jitter * (1 - RandomProvider.NextDouble())));
```

Comment:
Correct me if I'm wrong please, the formula shouldn't be like 
``` csharp 
var delay = _currentBackOffMilliseconds  + (int)(_currentBackOffMilliseconds * (1 - _jitter * (1 - RandomProvider.NextDouble())));
```
I mean delay should be >= `_currentBackOffMilliseconds`. In this case, formula takes `_currentBackOffMilliseconds*n where  0<=n<=1` and less than `_currentBackOffMilliseconds`.